polaris-react/src/components/IndexTable/IndexTable.scss

```
@@ -220,83 +191,62 @@ $loading-panel-height: 53px;
     .TableCell-first,
     .TableCell-first + .TableCell {
       background-color: var(--p-color-bg-subdued);
-      #{$se23} & {
-        color: var(--p-color-text-subdued);
-      }
+      color: var(--p-color-text-subdued);
     }
   }
 
-  &.TableRow-hovered {
+  &.TableRow-selected {
+    border-color: var(--p-color-border);
     // stylelint-disable-next-line selector-max-class, selector-max-combinators, selector-max-specificity -- generated by polaris-migrator DO NOT COPY
     &,
+    .TableHeading-first,
+    .TableHeading-second,
     .TableCell-first,
     .TableCell-first + .TableCell {
-      background-color: var(--p-color-bg-hover);
+      background-color: var(--p-color-bg-primary-subdued-selected);
     }
   }
 
-  &.TableRow-selected {
-    // stylelint-disable-next-line selector-max-class, selector-max-combinators, selector-max-specificity -- generated by polaris-migrator DO NOT COPY
+  &.TableRow-hovered {
```

The order of the cascade here is particularly important, so I'm paraphrasing the comment we had below in the se23 styles:

```suggestion
  // NOTE: `-hovered` must come after `-selected` so elements don't appear to lose their interactivity when selected.
  &.TableRow-hovered {
```